src/Tasks/ResolveManifestFiles.cs

```diff
@@ -680,15 +686,16 @@ private bool IsFiltered(ITaskItem item)
             // OpenScope and returns null if not an assembly, which is much faster.
 
             AssemblyIdentity identity = AssemblyIdentity.FromManagedAssembly(item.ItemSpec);
-            if (item.ItemSpec.EndsWith(".dll") && identity == null)
+            if (item.ItemSpec.EndsWith(".dll") && identity == null && !isDotNetCore)
             {
                 // It is possible that a native dll gets passed in here that was declared as a content file
-                // in a referenced nuget package, which will yield null here. We just need to ignore those, 
-                // since those aren't actually references we care about.
+                // in a referenced nuget package, which will yield null here. We just need to ignore those 
+                // for .NET FX case since those aren't actually references we care about. For .NET Core, native
+                // dll can be passed as a reference so we won't ignore it if isDotNetCore is true.
```

Comment:
Can you elaborate about why we want this for core and not full fx?

Reply:
IsFiltered() is called to filter out References that we don't want to publish in ClickOnce.

For Full FX, we haven't seen any native references (up until 4.8 at least) in References group and don't want to change the behavior for back compat reason. 
For .NET Core apps, we are seeing packages with native DLLs in runtime assets that we need to publish and they show up in the References item group. So I am adding exclusion for .NET Core apps when we filter native DLLs.

Reply:
For what it's worth, both SDK style projects targeting the .NET Framework, and also non-SDK style projects can have native assemblies come from a runtimes folder in a package. For non-SDK style projects, it's less likely, as the consuming project may need to add another package reference to bring in a `runtimes.json` file so that NuGet knows how to map RIDs correctly. But it's possible.

Reply:
OK that's good to know. Since ClickOnce for Full FX has always ignored native references for those projects, we're going to retain the behavior and only allow native references to be pushished for .NET Core projects.